File: src/main/java/com/pygostylia/osprey/ChunkPosition.java
package com.pygostylia.osprey;

public record ChunkPosition(int x, int z) {
    public Double distanceFrom(ChunkPosition location) {
        return Math.sqrt(
                Math.pow(Math.abs(location.x - x), 2) +
                        Math.pow(Math.abs(location.z - z), 2)
        );
    }
}

File: src/main/java/com/pygostylia/osprey/ChunkPosition.kt
package com.pygostylia.osprey

import kotlin.math.*

data class ChunkPosition(val x: Int, val z: Int) {
    fun distanceFrom(c: ChunkPosition): Double {
        return sqrt(
            abs(c.x - x).toDouble().pow(2.0) + abs(c.z - z).toDouble().pow(2.0)
        )
    }
}